executor/runtime/jobmaster/build.go

```diff
@@ -18,6 +30,44 @@ func RegisterBuilder() {
 
 type jobMasterBuilder struct{}
 
+func getEtcdMetaKV(ctx context.Context, clients *client.Manager) (metadata.MetaKV, error) {
+	resp, err := clients.MasterClient().QueryMetaStore(
+		ctx,
+		&pb.QueryMetaStoreRequest{Tp: pb.StoreType_ServiceDiscovery},
+		5*time.Second,
+	)
+	if err != nil {
+		return nil, err
+	}
+	log.L().Info("update service discovery metastore", zap.String("addr", resp.Address))
+
+	logConfig := logutil.DefaultZapLoggerConfig
+	logConfig.Level = zap.NewAtomicLevelAt(zapcore.ErrorLevel)
+	etcdCli, err := clientv3.New(clientv3.Config{
+		Endpoints:   strings.Split(resp.GetAddress(), ","),
+		Context:     ctx,
+		LogConfig:   &logConfig,
+		DialTimeout: 5 * time.Second,
+		DialOptions: []grpc.DialOption{
+			grpc.WithInsecure(),
+			grpc.WithBlock(),
+			grpc.WithConnectParams(grpc.ConnectParams{
+				Backoff: backoff.Config{
+					BaseDelay:  time.Second,
+					Multiplier: 1.1,
+					Jitter:     0.1,
+					MaxDelay:   3 * time.Second,
+				},
+				MinConnectTimeout: 3 * time.Second,
+			}),
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return metadata.NewMetaEtcd(etcdCli), nil
+}
+
 func (b *jobMasterBuilder) Build(op model.Operator) (runtime.Operator, bool, error) {
```

Comment:
what's the purpose of this file ?  if used to help create jobmaster, this file should be put under the jobmaster floder

Reply:
This file is set to analyze the incomming task/jobmaster config and build a runtime operator that drived by task.